src/awkward/_connect/numba/arrayview.py

```diff
@@ -291,6 +301,9 @@ def lower_const_view(context, builder, viewtype, view):
     proxyout.pylookup = context.add_dynamic_addr(
         builder, id(lookup), info=str(type(lookup))
     )
+    proxyout.pyattrs = context.add_dynamic_addr(
+        builder, id(attrs), info=str(type(attrs))
+    )
```

Comment:
One request: for the Numba implementation (not including ArrayBuilder), could you pass the `attrs` through by just attaching the `attrs` to the `lookup`? These are both Python objects, which is what makes them cumbersome to pass through lowered code, but we should only have to do that once. Every time the `lookup`'s `PyObject*` is copied, it slows down iteration in Numba-compiled functions. There is a one-to-one relationship between the `lookup` and the original array (the `lookup` is both holding the original array's references and also presenting them in a pointer/bytes-friendly form), so an array's `attrs` can be simply attached to its `lookup`.

Also, this `add_dynamic_addr` puts a pointer into the space of `static` variables. That's probably limited, so there's an implicit upper limit on the number of arrays that can be passed through Numba-compiled functions. I don't know what that limit is, but by making two pointers for every array, we approach it twice as fast.

In the boxing function, you'd be able to access the `attrs_obj` by

```python
c.pyapi.object_getattr_string(lookup_obj, "_attrs")
```

Oddly enough, we're serializing and unserializing the entire `behaviors` dict and that seems to be okay. It seems to me that the `behaviors` could be carried on the `lookup` as well. I'm surprised that no one in Coffea ran into the issue of putting a lambda function in `behaviors` and then not being able to pass it through a Numba-compiled function.